docs/articles/documentation/guides/advanced-guides/test-https-features-and-http2-websites.md

```diff
@@ -23,21 +25,107 @@ If the tested page does not use HTTPS-specific features (like
+[SubtleCrypto](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto)), TestCafe successfully proxies it over HTTP. Otherwise, the tests fail because the page throws JavaScript errors, does not load or loads partially.
 
-### Proxy the Website Over HTTPS
+To establish a secure HTTPS-connection, Node.js requires a valid [PKCS12](https://en.wikipedia.org/wiki/PKCS_12) certificate (`.pfx`). If you have a certificate signed by a trusted authority, we recommend that you [use this certificate](#use-a-trusted-certificate) if your security policy permits. The browsers are then able to open the tested page without workarounds. Alternatively, you can [generate and use a self-signed certificate](#use-a-self-signed-certificate).
 
-When you run tests from the command line, use the [--ssl](../../reference/command-line-interface.md#--ssl-options) flag to enable HTTPS on a proxy server. Specify options required to initialize [a Node.js HTTPS server](https://nodejs.org/api/https.html#https_https_createserver_options_requestlistener) after this flag in a semicolon-separated string. The most commonly used SSL options are described in the [TLS topic](https://nodejs.org/api/tls.html#tls_tls_createsecurecontext_options) in the Node.js documentation.
+> Before you try HTTPS, launch testcafe with [`--hostname localhost` CLI parameter](../../reference/command-line-interface.md#--hostname-name) or [`hostname: localhost` configuration option](../../reference/configuration-file.md#hostname). Since all modern browsers thread localhost as a secure origin, some built-in browser features that require secure origin (Service workers, GeoAPI) will work. Third-party JS features like ApplePaySession may or may not work depending on implementation.
 
-The example below uses the PFX encoded private key and certificate chain to create an HTTPS server.
+### Use a Trusted Certificate
+
+To provide a certificate when you launch TestCafe with [CLI](../../reference/command-line-interface.md), use [`-ssl`](../../reference/command-line-interface.md#--ssl-options) option:
 
 ```sh
-testcafe --ssl pfx=path/to/file.pfx;rejectUnauthorized=true;...
+testcafe chrome  test.js --ssl pfx=path/to/trusted/certificate.pfx;rejectUnauthorized=true;
 ```
 
-When you use the programming interface, pass the HTTPS server options to the [createTestCafe](../../reference/testcafe-api/global/createtestcafe.md) method.
+### Use a Self-Signed Certificate
+
+[Generate a self-signed certificate with `openssl`](#generate-a-certificate-from-the-command-line) if you launch tests from the command line. In the API, you can use the [openssl-self-signed-certificate](https://www.npmjs.com/package/openssl-self-signed-certificate) module to [generate a certificate in code](#generate-a-certificate-in-your-code).
```

Comment:
You already mention "In the API", why do you then add  "in code"?

Reply:
I want the link text to be descriptive

Reply:
To generate a certificate in code, use the openssl-self-signed-certificate module.